### PR TITLE
feat: new reviews module

### PR DESCRIPTION
**Pull request by evan-spidy**

### CODE REVIEW

File: src/components/ForYou/index.tsx [hunk 1]
```diff
@@ -159,6 +257,62 @@ const ForYou = () => {
...
+            Select Your Areas of Interest
+          </span>
+          <span className="block mt-[8px] mb-[24px] text-[9px] leading-[10px]">
+            Your preferences will help us create a journey unique to you.
+          </span>
+
+          <button className="mt-[24px] bg-primary text-white text-[14px] leading-[14px] font-outfit font-bold py-[10px] w-full rounded-[24px] mt-[24px] mb-[16px]">
+            Let's Begin
+          </button>
+        </Modal>
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: the code is repeated

File: src/components/ForYou/index.tsx [hunk 2]
```diff
@@ -159,6 +257,62 @@ const ForYou = () => {
...
+          <div className="relative w-full flex items-center justify-center py-[17px] px-[19px]">
+            <span className="font-outfit text-[16px] leading-[1] font-bold text-white">
+              Reviews
+            </span>
+            <i
+              className="votigram-icon-cancel text-[14px] text-white absolute top-[15px] right-[18px] cursor-pointer"
+              onClick={() => setIsShowReviews(false)}
+            />
+          </div>
+          <div className="py-[4px]">
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: this can replace with py-1

File: src/components/ForYou/index.tsx [hunk 3]
```diff
@@ -159,6 +257,62 @@ const ForYou = () => {
...
+            <List
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: what list is this? can you use a better name?

File: src/components/ForYou/index.tsx [hunk 4]
```diff
@@ -159,6 +257,62 @@ const ForYou = () => {
...
+              hasMore={hasMoreData}
+              height="60vh"
+              dataSource={allItems}
+              loadData={() => setSize(size + 1)}
+              emptyText="Write the first review!"
+              rootClassname="px-[20px]"
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: this can be use as px-5

File: src/components/ForYou/index.tsx [hunk 5]
```diff
@@ -159,6 +257,62 @@ const ForYou = () => {
...
+                  <div className="flex items-center justify-center h-full">
+                    <div className="w-[30px] h-[30px] animate-spin rounded-full bg-white"></div>
+                  </div>
+                )
+              }
+            />
+          </div>
+          <div className="py-[17px] px-[20px] border-t-[1px] border-tertiary border-solid">
+            <Textarea
+              disabled={loadding}
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: check your spelling

File: src/components/List/components/Item.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any;
+  className?: string;
+  onClick?: (item: ListItem) => void;
+}
+
+const Item = ({ data, className, onClick }: ItemProps) => {
+  return (
+    <div
+      className={`flex flex-row items-start gap-[19px] py-[9px] ${className}`}
```
Comment (evan-spidy) [inline src/components/List/components/Item.tsx]: use clsx

File: src/components/List/components/Item.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      onClick={() => onClick?.(data)}
+    >
+      {data.commenterPhoto && (
```
Comment (evan-spidy) [inline src/components/List/components/Item.tsx]: check data? is undefined

File: src/components/List/components/Item.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        <div className="w-[29px] h-[29px] rounded-[47.5px] bg-tertiary shrink-0 overflow-hidden">
+          <img
+            src={data.commenterPhoto}
+            alt="Avatar"
+            className="w-full h-full object-cover"
+          />
+        </div>
```
Comment (evan-spidy) [inline src/components/List/components/Item.tsx]: this is not nextjs, you can use img will dop

File: src/components/List/index.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  return (
+    <div
+      ref={listRef}
+      role="list"
+      className="overflow-x-hidden overflow-y-auto"
+      style={{ height: calculatedHeight }}
+    >
+      {items.length === 0 && (
+        <div className="flex items-center justify-center h-full">
+          <span className="font-normal text-[13px] text-white font-questrial leading-[1.2]">
```
Comment (evan-spidy) [inline src/components/List/index.tsx]: do not need to add questrial

File: src/components/Textarea/index.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        <textarea
+          ref={textareaRef}
+          className={clsx(
+            "p-0 text-[14px] leading-[16.8px] outline-none resize-none overflow-hidden appearance-none bg-input",
+            rootClassName
+          )}
+          value={text}
+          maxLength={maxLength}
+          onChange={handleChange}
+          placeholder={placeholder || "Please enter ..."}
```
Comment (evan-spidy) [inline src/components/Textarea/index.tsx]: are you suppose to have a space between 'enter' and '...' ?

File: src/components/ForYou/index.tsx (hunk 1 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

File: src/components/ForYou/index.tsx (hunk 3 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

File: src/components/ForYou/index.tsx (hunk 4 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

File: src/components/ForYou/index.tsx (hunk 5 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

File: src/components/List/components/Item.tsx (hunk 6 above)
Comment (Leon-pix-coder) [reply src/components/List/components/Item.tsx]: Updated

File: src/components/List/components/Item.tsx (hunk 7 above)
Comment (Leon-pix-coder) [reply src/components/List/components/Item.tsx]: Updated

File: src/components/List/components/Item.tsx (hunk 8 above)
Comment (Leon-pix-coder) [reply src/components/List/components/Item.tsx]: Updated

File: src/components/Textarea/index.tsx (hunk 10 above)
Comment (Leon-pix-coder) [reply src/components/Textarea/index.tsx]: Updated

File: src/components/ForYou/index.tsx (hunk 2 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

File: src/components/List/index.tsx (hunk 9 above)
Comment (Leon-pix-coder) [reply src/components/List/index.tsx]: Removed

**Pull request merged**